pkg/driver/controller_server.go

```diff
@@ -758,7 +779,7 @@ func (driver *Driver) ControllerPublishVolume(ctx context.Context, request *csi.
 		publishContext[chapUsername] = publishInfo.AccessInfo.BlockDeviceAccessInfo.IscsiAccessInfo.ChapUser
 		publishContext[chapPassword] = publishInfo.AccessInfo.BlockDeviceAccessInfo.IscsiAccessInfo.ChapPassword
 	}
-	publishContext[readOnly] = strconv.FormatBool(request.Readonly)
```

Comment:
I think we shouldn't completely the 'request.Readonly' parameter value here. 
Instead we could say as below:

if readOnlyAccessMode == "true {
           publishContext[readOnly] = strconv.FormatBool(readOnlyAccessMode)
} else { // Default case, we stick to old behavior
           publishContext[readOnly] = strconv.FormatBool(request.Readonly)
}

Reply:
i agree, this might be better in case if they fix the request.ReadOnly with Controller Publish it will continue to work..